src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -318,6 +327,178 @@ private void VerifyDiagnosticResults(IEnumerable<Diagnostic> actualResults, Immu
             }
         }
 
+        private ImmutableArray<(Diagnostic? actual, DiagnosticResult? expected)> MatchDiagnostics(Diagnostic[] actualResults, DiagnosticResult[] expectedResults)
+        {
+            expectedResults = expectedResults.ToOrderedArray();
+
+            var bestMatchCount = actualResults.Length + expectedResults.Length;
+            var bestMatch = actualResults.Select(result => ((Diagnostic?)result, default(DiagnosticResult?))).Concat(expectedResults.Select(result => (default(Diagnostic?), (DiagnosticResult?)result))).ToImmutableArray();
+
+            var builder = ImmutableArray.CreateBuilder<(Diagnostic? actual, DiagnosticResult? expected)>();
+            bool[] usedExpected = new bool[expectedResults.Length];
+            RecursiveMatch(0, 0, 0, usedExpected);
```

Comment:
You might want to add a discard or a comment to make it explicit you want to ignore the result.

Reply:
Done!